lib/beyond_api/services/product_management/product.rb

```diff
@@ -116,6 +116,258 @@ def create(body)
+      def update_product(id, body)
+        patch("products/#{id}", body)
+      end
+
+      # Update multiple products partially (Currently, this endpoint can only be used to update the visible property).
+      #
+      # @see https://developer.epages.com/beyond-docs/#update_multiple_products_partially
+      #
+      # @param ids [Array<String>] the IDs of the products to update
+      # @param visible [Boolean] visibility status to apply to all products
+      #
+      # @return [Hash]
+      #
+      # @example
+      #   @client.update_products_visibility(['a32ef424-ecc4-4bfc-815b-4f15bb3cffa1', 'a98292eb-0c64-459b-b28d-e229013b86b7'], true)
+      def update_products_visibility(ids, visible)
+        body = {
+          ids:,
+          details: {
+            visible:
+          }
+        }
+        patch('products/batch', body)
+      end
+
+      # Update a variation product partially.
+      #
+      # @see https://developer.epages.com/beyond-docs/#update_variation_product_partially
+      #
+      # @param id [String] the UUID of the variation product
+      # @param body [Hash] the request body with fields to update
+      #
+      # @return [Hash]
+      #
+      # @example
+      #   variation_update_data = {
+      #     name: 'Rioja Castillo de Puerto (2013)',
+      #     description: 'Spain\nRioja Tempranillo',
+      #     variationAttributes: [
+      #       {
+      #         displayName: 'Width',
+      #         values: ['100', '150'],
+      #         variationImagesDifferentiator: false
+      #       }
+      #     ],
+      #     manufacturer: 'Grape Vineyard',
+      #     salesPrice: {
+      #       taxModel: 'GROSS',
+      #       currency: 'EUR',
+      #       amount: 8.7
+      #     }
+      #   }
+      #   @client.update_variation_product('e0c285ad-d728-475a-974e-cd5360ceeb4f', variation_update_data)
+      def update_variation_product(id, body)
+        patch("products/#{id}", body)
+      end
+
+      # Delete a product or variation product.
+      #
+      # @see https://developer.epages.com/beyond-docs/#delete_product
+      #
+      # @param id [String] the UUID of the product to delete
+      #
+      # @return [nil]
+      #
+      # @example
+      #   @client.delete_product('4f8c204e-b56e-4e10-93b5-24ac56df2db0')
+      def delete_product(id)
+        delete("products/#{id}")
+      end
+
+      # Delete multiple products or variation products.
+      #
+      # @see https://developer.epages.com/beyond-docs/#delete_multiple_products
+      #
+      # @param ids [Array<String>] the IDs of the products to delete
+      #
+      # @return [nil]
+      #
+      # @example
+      #   @client.delete_multiple_products(['e6f1d2ef-df46-4d99-bbca-550971b51c17', 'b9fcf3d7-4e76-4818-9f48-2ce43929d35b'])
+      def delete_multiple_products(ids)
+        body = { ids: }
+        delete('products/batch', {}, body) # Mirar igual como estaba hecho en beyond api antiguo
```

Comment:
Don't forget to delete this spanish comment. It's always helpful to add a # TODO: and also a git hook to prevent this things to happen.

Reply:
Good one 👍🏼